Add func TestData

yitopic opened the pull request:
In addition to add testing.go and testing_test.go, I also renamed candy.go into file.go and candy_test.go into file_test.go.

yitopic committed:
Add testing.go and unit test

yitopic committed:
Rename candy.go into file.go

yitopic committed:
More comments

yitopic committed:
No longer export MustOpen and MustCreate, because users might forget to close the opened/created files

ghost commented:
Make MustOpen and MustCreate no longer exported, since users might forget to close the opened/created files.

cwtopic commented:
is there a reason/convention to use lower case mustOpen and mustCreate? 

ah, i think mustOpen won't be used by other programs by its naming convention.

@cwtopic Yes. You got it!

cwtopic commented on file.go:
```diff
@@ -38,7 +36,7 @@ func MustCreate(fileanme string) io.WriteCloser {
 //            return s
 //    }).(*AStruct)
 func OpenAnd(file string, fn func(r io.Reader) interface{}) interface{} {
```
a bit confusing name. OpenAnd only allows reading. What about appending the same file?

yitopic replied:
It is a Go convention since Go standard package `os` provides `os.Open` to open a file for reading, and `os.Create` to open (create) a file only for writing.  Both `os.Open` and `os.Create` calls `os.OpenFile`, which is the Go equivalent to C's `fopen` and can open for reading and/or writing.

How about rename `OpenAnd` to be `WithOpened` and rename `CreateAnd` to be `WithCreated`?

yitopic committed:
Rename OpenAnd into WithOpened; Rename CreateAnd into WithCreated; Add ReadAll

yitopic commented:
I renamed `OpenAnd` into `WithOpened`; renamed `CreateAnd` into `WithCreated`. I also added a `ReadAll` which returns a `bytes.Reader` object which implements the `io.Reader` interface.  How do you think about these changes? :-)

brjg commented:
I think it's good:)
On Sat, Jan 30, 2016 at 10:54 PM yitopic notifications@github.com wrote:

> I renamed OpenAnd into WithOpened; renamed CreateAnd into WithCreated. I
> also added a ReadAll which returns a bytes.Reader object which implements
> the io.Reader interface. How do you think about these changes? :-)
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/topicai/candy/pull/1#issuecomment-177415011.
